fedimint-server/src/config/api.rs

```diff
@@ -755,31 +811,44 @@ mod tests {
                 p2p_url,
                 api_url: api_url.clone(),
             };
-            let server = run_server(
-                data_dir.clone(),
-                our_connections.clone(),
-                db,
-                Default::default(),
-                Default::default(),
-                Default::default(),
-            )
-            .await;
+            let dir = data_dir.join(name_suffix.to_string());
+            fs::create_dir_all(dir.clone()).expect("Unable to create test dir");
+
+            let connections = our_connections.clone();
+            let mut task_group = TaskGroup::new();
+            let subgroup = task_group.make_subgroup().await;
+            task_group
+                .spawn("run-server", move |_| async move {
+                    run_server(
```

Comment:
The number of arguments calls for either a builder (especially given the amount of `Default::default`, or at least packing some of them into some aggregating `struct`.

Reply:
Good point, I'll refactor